src/play_controller.cpp

```diff
@@ -280,7 +281,59 @@ void play_controller::init(const config& level)
 		plugins_context_->set_callback("save_game", [this](const config& cfg) { save_game_auto(cfg["filename"]); }, true);
 		plugins_context_->set_callback("save_replay", [this](const config& cfg) { save_replay_auto(cfg["filename"]); }, true);
 		plugins_context_->set_callback("quit", [](const config&) { throw_quit_game_exception(); }, false);
+		plugins_context_->set_callback_execute(*resources::lua_kernel, true);
 		plugins_context_->set_accessor_string("scenario_name", [this](config) { return get_scenario_name(); });
+		plugins_context_->set_accessor_int("current_side", [this](config) { return current_side(); });
+		plugins_context_->set_accessor_int("current_turn", [this](config) { return turn(); });
+		plugins_context_->set_accessor_bool("can_move", [this](config) { return !events::commands_disabled && gamestate().gamedata_.phase() == game_data::TURN_PLAYING; });
+		plugins_context_->set_callback("end_turn", [this](config) { force_end_turn(); }, false);
```

Comment:
force_end_turn() was meant to be called from syned contexts, idk whetehr it actually cases OOS, but i think it better to juse use require_end_turn() here